### DISCUSSION TITLE
.heic full container upload

### DISCUSSION THREAD

**Discussion by T3ll1nG** [ente-io/ente#2572] [General]
I appreciate that the ente app currently splits Apple Live Photos into separate .heic (still image) and .mov (video) files for display purposes. While this works well within the app, it creates challenges for users who want to use it as a true backup solution or modify the photos later (e.g., changing the key image).

To address this, I propose adding an option to upload the original .heic container file, which includes both the still image and live video components. This would provide users with a complete backup of the original Live Photo file. I suggest the following implementation:

- Add a setting to upload the original .heic container file.
- Make this optional, as users may prefer the current split-file approach.
- Allow users to choose whether to upload both the split files and the original container.
- Provide an option in the export function to export only the original .heic file.

I understand that choosing to upload only the original .heic container might result in losing the ability to play the video portion within the app. However, this trade-off would be acceptable for users prioritizing a true backup of their original files.

This feature would eliminate the need for users to rely on additional cloud services to maintain copies of their original Live Photo files.

**Comment by ua741** [comment]
From our understanding, Apple Photos themself maintain two separate file, that's what we get via API.
If you try to export a Live photo from either Apple or Google, you will get two separate files.

Do you have a reference/documentation link which talks about `.heic  container` for Apple Live photos?

**Comment by T3ll1nG** [comment]
Hi @ua741, 
sorry this is totally my fault. This is a none issue and can be closed. 
I must have done something wrong on the import, tried it again and could not replicate what I saw before. Your solution works fine, iOS and macOS import it as a live photo. It didn't work for me initially and I read some faulty articles that said it was one container. Upon further inspection your solution is absolutely the way it is done, this are 2 files that are linked in the DB if you use photos and the export will give you the one `.heic`and one `.mov`.

**Reply by ua741** [reply]
No worries, glad everything is sorted.
